File: 5_permutations.py
n = int(input())

if n == 1:
    print("1")
elif n == 3 or n == 2:
    print("NO SOLUTION")
else:
    for i in range(2, n + 1, 2):
        print(i)
    for i in range(1, n+1, 2):
        print(i)

File: 5_permutations.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
	int n;
	cin >> n;
	if(n == 1){
		cout << n;
		return 0;
	} 
	if((n == 2) or (n ==3)){
		cout << "NO SOLUTION";
		return 0;
	} 

	if(n%2 == 0){
		for(int i = 2; i <= n; i += 2){
			cout << i << " ";
		}
		for(int i = 1; i < n; i+=2){
			cout << i << " ";
		}
	}else{
		for(int i = 2; i < n; i += 2){
			cout << i << " ";
		}
		for(int i = 1; i <= n; i+=2){
			cout << i << " ";
		}
	}
	
}